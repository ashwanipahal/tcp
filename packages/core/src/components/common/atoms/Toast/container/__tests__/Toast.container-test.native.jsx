import React from 'react';
import { shallow } from 'enzyme';
import ToastContainer, { mapDispatchToProps } from '../Toast.container.native';

describe('Toast render', () => {
  it('should render correctly', () => {
    const props = {
      errorMessage: 'errormessage',
    };
    const tree = shallow(<ToastContainer {...props} />);
    expect(tree).toMatchSnapshot();
  });
  describe('#mapDispatchToProps', () => {
    it('should return an action toastMessageReset which will call dispatch function on execution', () => {
      const dispatch = jest.fn();
      const dispatchProps = mapDispatchToProps(dispatch);
      dispatchProps.toastMessageReset();
      expect(dispatch.mock.calls).toHaveLength(1);
    });
  });
});
