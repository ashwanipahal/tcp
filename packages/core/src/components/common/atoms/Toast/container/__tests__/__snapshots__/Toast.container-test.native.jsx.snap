// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast render should render correctly 1`] = `
<ToastView
  errorMessage="errormessage"
  positionValue={1}
  toastMessageReset={[MockFunction]}
/>
`;
